pkgs/development/compilers/factor-lang/wrapper.nix

```
@@ -0,0 +1,143 @@
+{ lib
+, stdenv
+, makeWrapper
+, factor-unwrapped
+, guiSupport ? true
+, cairo
+, freealut
+, gdk-pixbuf
+, glib
+, gnome2
+, gtk2-x11
+, libGL
+, libGLU
+, librsvg
+, graphviz
+, coreutils
+, gnugrep
+, libogg
+, libvorbis
+, openal
+, openssl
+, pango
+, pcre
+, udis86
+, zlib
+  # Libraries added to ld.so.cache
+, runtimeLibs ? [ ]
+  # Packages added to path (and ld.so.cache)
+, binPackages ? [ ]
+  # Extra vocabularies added to out/lib/factor
+, extraVocabs ? [ ]
+  # Enable default libs and bins to run most of the standard library code.
+, enableDefaults ? true
+}:
+let
+  inherit (lib) optional optionals optionalString;
+  # missing from lib/strings
+  escapeNixString = s: lib.escape [ "$" ] (builtins.toJSON s);
+  toFactorArgs = x: lib.concatStringsSep " " (map escapeNixString x);
+  defaultLibs = [ libogg libvorbis openal openssl pcre udis86 zlib ];
+  defaultBins = [ graphviz ];
+  libs = (optionals enableDefaults defaultLibs) ++ (optionals guiSupport [
+    cairo
+    freealut
+    gdk-pixbuf
+    glib
+    gnome2.gtkglext
+    gtk2-x11
+    libGL
+    libGLU
+    pango
+  ]) ++ runtimeLibs ++ binPackages;
+  bins = binPackages ++ (optionals enableDefaults defaultBins);
+
+in
+stdenv.mkDerivation (finalArgs: {
+  pname = "${factor-unwrapped.pname}-env";
+  inherit (factor-unwrapped) version;
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = optional guiSupport gdk-pixbuf;
+
+  dontUnpack = true;
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
```

```suggestion
  dontUnpack = true;
```
those phases are automatically skipped if the files don't exist

---

Done.